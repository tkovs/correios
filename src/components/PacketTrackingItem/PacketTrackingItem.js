import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { View, StyleSheet } from 'react-native'
import { Caption, Text, Title } from 'react-native-paper'

function PacketTrackingItem({ date, status, from, to, locale, note }) {
  const time = moment(date).format('h:mm')
  const day = moment(date).format('D MMM')

  return (
    <View style={styles.container}>
      <View style={styles.left}>
        <Text>{time}</Text>
        <Text>{day}</Text>
      </View>
      <View style={styles.right}>
        <Title>{status}</Title>
        <Text>De: {from}</Text>
        {to ? <Text>Para: {to}</Text> : null}
        <Text>[gps icon] {locale}</Text>
        {note ? <Caption>Obs.: {note}</Caption> : null}
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: 'gray',
  },
  left: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRightWidth: 1,
    borderRightColor: 'gray',
    paddingRight: 8,
    marginRight: 8,
  },
})

PacketTrackingItem.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
  from: PropTypes.string,
  locale: PropTypes.string,
  note: PropTypes.string,
  status: PropTypes.string.isRequired,
  to: PropTypes.string,
}

PacketTrackingItem.defaultProps = {
  from: '',
  locale: '',
  note: '',
  to: '',
}

export default PacketTrackingItem
