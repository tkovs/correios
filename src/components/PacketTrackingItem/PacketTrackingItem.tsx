import React, { FC } from 'react'
import moment from 'moment'
import { View, StyleSheet } from 'react-native'
import { Avatar, Caption, Text } from 'react-native-paper'
import isEmpty from 'lodash/isEmpty'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { formatLocation, formatInitialLocation } from '../../utils/location'
import { colors } from '../../styles/theme'

interface PacketTrackingItemProps {
  datetime: Date
  status: string
  note?: string
  from?: Locale
  to?: Locale
  locale?: Locale
}

type Props = PacketTrackingItemProps

const PacketTrackingItem: FC<Props> = ({
  datetime,
  status,
  from,
  to,
  locale,
  note,
}: Props) => {
  const time = moment(datetime).format('h:mm')
  const day = moment(datetime).format('D MMM')

  return (
    <View style={styles.container}>
      <View style={styles.left}>
        <Text
          accessibilityStates={['disabled']}
          style={[styles.textColor, styles.alignTextToLocation]}
        >
          {time}
        </Text>
        <Text
          accessibilityStates={['disabled']}
          style={[styles.textColor, styles.alignTextToLocation]}
        >
          {day}
        </Text>
      </View>
      <View style={styles.icon}>
        <Avatar.Icon
          accessibilityStates={['disabled']}
          size={48}
          icon="truck"
          color={colors.blue}
        />
      </View>
      <View style={styles.right}>
        <Text
          accessibilityStates={['disabled']}
          style={[
            styles.textColor,
            styles.alignTextToLocation,
            styles.boldText,
          ]}
          numberOfLines={1}
        >
          {status}
        </Text>
        {!isEmpty(from) ? (
          <Text
            accessibilityStates={['disabled']}
            style={[styles.textColor, styles.alignTextToLocation]}
          >
            De: {formatLocation(from)}
          </Text>
        ) : null}
        {!isEmpty(to) ? (
          <Text
            accessibilityStates={['disabled']}
            style={[styles.textColor, styles.alignTextToLocation]}
          >
            Para: {formatLocation(to)}
          </Text>
        ) : null}
        {!isEmpty(locale) ? (
          <Text
            accessibilityStates={['disabled']}
            style={[styles.textColor, styles.alignTextToLocation]}
          >
            De: {formatLocation(locale)}
          </Text>
        ) : null}
        <View style={styles.locationInfo}>
          <Icon name="location-on" size={20} color={colors.licorice} />
          <Text
            accessibilityStates={['disabled']}
            style={[styles.textColor, styles.alignTextToLocation]}
          >
            {formatInitialLocation({ from, to, locale })}
          </Text>
        </View>
        {note ? <Caption>Obs.: {note}</Caption> : null}
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 16,
    paddingRight: 16,
    flexDirection: 'row',
  },
  icon: {
    justifyContent: 'center',
    paddingRight: 4,
  },
  left: {
    justifyContent: 'center',
    alignItems: 'center',
    minWidth: 64,
  },
  textColor: {
    color: colors.licoric,
  },
  alignTextToLocation: {
    paddingLeft: 4,
  },
  right: {},
  locationInfo: {
    flexDirection: 'row',
  },
  boldText: {
    fontWeight: 'bold',
  },
})

PacketTrackingItem.defaultProps = {
  note: '',
  from: undefined,
  locale: undefined,
  to: undefined,
}

export default PacketTrackingItem
