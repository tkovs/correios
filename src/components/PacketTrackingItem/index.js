import React from 'react'
import PropTypes from 'prop-types'

import PacketTrackingItem from './PacketTrackingItem'
import { apiDateTimeToDate } from '../../utils/datetime'

function PacketTrackingItemContainer({
  datetime: dt,
  status,
  from,
  to,
  locale,
}) {
  const datetime = apiDateTimeToDate(dt)

  return (
    <PacketTrackingItem
      status={status}
      datetime={datetime}
      from={from}
      to={to}
      locale={locale}
      note=""
    />
  )
}

PacketTrackingItemContainer.propTypes = {
  status: PropTypes.string.isRequired,
  datetime: PropTypes.shape({
    date: PropTypes.shape({
      day: PropTypes.number,
      month: PropTypes.number,
      year: PropTypes.number,
    }),
    time: PropTypes.shape({
      hour: PropTypes.number,
      minute: PropTypes.number,
    }),
  }).isRequired,
  from: PropTypes.shape({
    city: PropTypes.string,
    state: PropTypes.string,
    place: PropTypes.string,
  }),
  to: PropTypes.shape({
    city: PropTypes.string,
    state: PropTypes.string,
    place: PropTypes.string,
  }),
  locale: PropTypes.shape({
    city: PropTypes.string,
    state: PropTypes.string,
    place: PropTypes.string,
  }),
}

PacketTrackingItemContainer.defaultProps = {
  from: undefined,
  to: undefined,
  locale: undefined,
}

export default PacketTrackingItemContainer
