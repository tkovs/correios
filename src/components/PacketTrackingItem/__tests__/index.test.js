import React from 'react'
import { render } from '@testing-library/react-native'

import PacketTrackingItem from '..'

const mockDatetime = new Date('2020-01-05T00:01:01')
const mockFrom = {
  place: 'AC ASSIS CHATEAUBRIAND',
  city: 'Assis Chateaubriand',
  state: 'PR',
}
const mockTo = {
  place: 'CTCE CURITIBA',
  city: 'Curitiba',
  state: 'PR',
}
const mockLocale = {
  place: 'CTCE CURITIBA',
  city: 'Curitiba',
  state: 'PR',
}

describe('PacketTrackingItem', () => {
  it('should render correctly a posted status', () => {
    const mockStatus = 'Objeto postado após o horário limite da unidade'
    const { baseElement } = render(
      <PacketTrackingItem
        datetime={mockDatetime}
        locale={mockLocale}
        status={mockStatus}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should render correctly an on course object', () => {
    const mockStatus = 'Objeto encaminhado'
    const { baseElement } = render(
      <PacketTrackingItem
        datetime={mockDatetime}
        from={mockFrom}
        status={mockStatus}
        to={mockTo}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should render correctly a delivered object', () => {
    const mockStatus = 'Objeto entregue ao destinatário'
    const { baseElement } = render(
      <PacketTrackingItem
        datetime={mockDatetime}
        locale={mockLocale}
        status={mockStatus}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
