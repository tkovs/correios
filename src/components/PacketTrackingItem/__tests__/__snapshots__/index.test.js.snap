// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacketTrackingItem should render correctly a delivered object 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          12:01
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          5 Jan
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Objeto entregue ao destinatário
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          De: 
          CTCE CURITIBA - Curitiba / PR
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PacketTrackingItem should render correctly a posted status 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          12:01
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          5 Jan
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Objeto postado após o horário limite da unidade
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          De: 
          CTCE CURITIBA - Curitiba / PR
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PacketTrackingItem should render correctly an on course object 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": 8,
            "paddingRight": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          12:01
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          5 Jan
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Objeto encaminhado
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          De: 
          AC ASSIS CHATEAUBRIAND - Assis Chateaubriand / PR
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Para: 
          CTCE CURITIBA - Curitiba / PR
        </Text>
      </View>
    </View>
  </View>
</View>
`;
